src/pages/index.js

```diff
@@ -1,140 +1,250 @@
 
 const popupWithImage = new PopupWithImage("#image-modal");
 popupWithImage.setEventListeners();
 
-const profileEditPopup = new PopupWithForm(
+const deleteImageModal = new PopupDeleteCard("#delete-card-modal");
+deleteImageModal.setEventListeners();
+
+const profileEditModal = new PopupWithForm(
   "#profile-edit-modal",
   (formData) => {
-    const title = formData[".profile-title-input"];
-    const description = formData[".profile-description-input"];
-    profileTitle.textContent = title;
-    profileDescription.textContent = description;
-    profileEditPopup.close();
+    handleEditProfileSubmit(formData);
   }
 );
-profileEditPopup.setEventListeners();
 
+profileEditModal.setEventListeners();
+
+//Declare API class:
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "2db58c83-3523-4135-8564-2660432887a4",
+    "Content-Type": "application/json",
+  },
+});
+
+//Declare UserInfo Class:
+const userInfo = new UserInfo({
+  titleSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  profileImage: ".profile__image",
+});
+
+//Declare PopupWithForm for addCardModal:
 const addCardPopup = new PopupWithForm("#add-card-modal", (formData) => {
-  const name = formData["input-type-title"];
-  const link = formData["input-type-url"];
-  renderCard(name, link);
-  addCardPopup.close();
+  handleAddCardSubmit(formData);
 });
 addCardPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const card = new Card(cardData, "#card-template", handleImageClick);
-      cardSection.addItem(card.getCardElement());
-    },
-  },
-  ".cards__list"
+//declare Edit Image Modal
+const profileImageModal = new PopupWithForm(
+  "#profile-image-modal",
+  (formData) => {
+    handleChangeProfilePicture(formData.url);
+  }
 );
+profileImageModal.setEventListeners();
 
-cardSection.renderItems();
+//Declare Section for Initial Cards
+let cardSection;
 
+//RenderCardFunction
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  cardWrap.prepend(card.getCardElement());
+  const card = createCard(cardData);
+  cardSection.addItem(card);
 }
-
+//Create Card Function
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getCardElement();
+}
+//Click Card Function
 function handleImageClick(link, name) {
   popupWithImage.open({ name, link });
 }
 
-// Form Validators
-const profileEditValidator = new FormValidator(config, profileEditForm);
-const addCardValidator = new FormValidator(config, addCardFormElement);
-profileEditValidator.enableValidation();
-addCardValidator.enableValidation();
+function handleEditProfileSubmit(inputValues) {
+  profileEditModal.renderLoading("Saving...");
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```